internal/handlers/bot.go

```diff
@@ -73,28 +86,113 @@ type IncomingEvent struct {
 }
 
+func loadRichMessage() (json.RawMessage, error) {
+	data, err := os.ReadFile("rich_message.json")
+	if err != nil {
+		return nil, err
+	}
+	return json.RawMessage(data), nil
+}
+
 func (h *WebhookHandler) Reply(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusAccepted)
-
 	var webhook Webhook
 
 	if err := json.NewDecoder(r.Body).Decode(&webhook); err != nil {
 		log.Printf("Error decoding webhook JSON: %v", err)
 		return
 	}
-
 	defer r.Body.Close()
 
 	log.Printf("Received webhook: %s, payload %s", webhook.Action, string(webhook.Payload))
 
 	switch {
 	case webhook.Action == "incoming_event":
-		log.Printf("Incoming envet action")
+		if err := h.handleIncomingEvent(webhook); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 	case webhook.Action == "incoming_chat":
-		log.Printf("Incoming chat action")
+		if err := h.handleIncomingChat(webhook); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 	default:
 		log.Printf("Webhook not recognized: %v", webhook)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *WebhookHandler) handleIncomingEvent(webhook Webhook) error {
+	var incomingEvent IncomingEvent
+	if err := json.Unmarshal(webhook.Payload, &incomingEvent); err != nil {
+		return fmt.Errorf("failed to unmarshall incoming event payload: %w", err)
+	}
+	token, exists := h.botStore.GetBotToken(webhook.OrganizationID)
+	if !exists {
+		return fmt.Errorf("bot token does not exists for organizationID: %s", webhook.OrganizationID)
+	}
+	var postback Postback
+	if err := json.Unmarshal(incomingEvent.Event.Postback, &postback); err != nil {
+		return fmt.Errorf("failed to unmarshal postback: %w", err)
+	}
+	log.Printf("Postback ID- %s", postback.ID)
+	switch postback.ID {
+	case "transfer_to_agent":
+		return h.handleTransferToAgent(incomingEvent, token)
+	case "continue_chat_with_bot":
+		return h.handleContinueWithBot(incomingEvent, token)
+	default:
+		return fmt.Errorf("Postback ID not configured %s", postback.ID)
+	}
+	return nil
+}
+
+func (h *WebhookHandler) handleTransferToAgent(event IncomingEvent, token string) error {
+	message := map[string]string{
+		"type": "message",
+		"text": "Understand. I will transfer chat to an agent now.",
+	}
+	if _, err := h.livechatSDK.SendEvent(event.ChatID, token, message); err != nil {
+		return fmt.Errorf("failed to send transfer message: %w", err)
+	}
+	if err := h.livechatSDK.TransferChat(event.ChatID, token); err != nil {
+		return fmt.Errorf("failed to transfer chat: %w", err)
+	}
+	return nil
+}
+
+func (h *WebhookHandler) handleContinueWithBot(event IncomingEvent, token string) error {
+	message := map[string]string{
+		"type": "message",
+		"text": "Super, How can I help you",
+	}
+	if _, err := h.livechatSDK.SendEvent(event.ChatID, token, message); err != nil {
+		return fmt.Errorf("failed to send continue message: %w", err)
+	}
```

Comment:
optional: just for the demonstration purposes, you could display another rich message in this scenario, with some sample scenarios for the customer, e.g. "book a visit", "cancel a visit", "reschedule a visit"